changelog.d/pa-3054.added

```diff
@@ -0,0 +1,38 @@
+Try-catch-else-finally is now supported in taint analysis.
+
+This change also includes some updates to our analysis. Previously we assumed that
+any statement inside the try clause may throw an exception, but now only
+function calls are assumed to possibly throw exceptions.
+
+Throw statements always throw an exception as it was before.
+
+This kind of statement is supported in languages including Python, Ruby, and Julia.
+
+Python example:
+
+```py
```

Comment:
After you removed `may_return` etc, please check if this example still behaves as you describe in the current PR.

Reply:
Yep!